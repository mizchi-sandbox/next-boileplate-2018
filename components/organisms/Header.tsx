import * as React from "react";

import { Link } from "../../routes";

export default () => (
  <div>
    <h1>Next.js boilerplate 2018 late</h1>
    <Link route="index">
      <a>index</a>
    </Link>
    |
    <Link route="about">
      <a>about</a>
    </Link>
    |
    <Link route="dynamic">
      <a>dynamic</a>
    </Link>
    |
    <Link route="lazy">
      <a>lazy</a>
    </Link>
    |
    <Link route="item" params={{ id: "a" }}>
      <a>/item/a</a>
    </Link>
    |
    <Link route="item" params={{ id: "b" }}>
      <a>/item/b</a>
    </Link>
    |
    <Link route="connected">
      <a>/connected</a>
    </Link>
  </div>
);
