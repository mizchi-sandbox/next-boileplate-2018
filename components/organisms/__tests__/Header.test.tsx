import * as React from "react";
import * as renderer from "react-test-renderer";
import Header from "../Header";

test("renders correctly", () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
