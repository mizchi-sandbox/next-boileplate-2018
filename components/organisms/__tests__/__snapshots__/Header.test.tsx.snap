// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Next.js boilerplate 2018 late
  </h1>
  <a
    href="/"
    onClick={[Function]}
  >
    index
  </a>
  |
  <a
    href="/about"
    onClick={[Function]}
  >
    about
  </a>
  |
  <a
    href="/dynamic"
    onClick={[Function]}
  >
    dynamic
  </a>
  |
  <a
    href="lazy"
    onClick={[Function]}
  >
    lazy
  </a>
  |
  <a
    href="/item/a"
    onClick={[Function]}
  >
    /item/a
  </a>
  |
  <a
    href="/item/b"
    onClick={[Function]}
  >
    /item/b
  </a>
  |
  <a
    href="/connected"
    onClick={[Function]}
  >
    /connected
  </a>
</div>
`;
